identityClient/identityClient_test.go

```diff
@@ -56,6 +57,275 @@ var (
+	}
+	queenClientInfo.Host = toznyCyclopsHost
+	identityServiceClient := New(queenClientInfo)
+	return identityServiceClient
+}
+
+func createIdentityServiceClientAndToken(t *testing.T) (E3dbIdentityClient, string) {
+	accountTag := uuid.New().String()
+	queenClientInfo, accountInfo, err := test.MakeE3DBAccount(t, &accountServiceClient, accountTag, toznyCyclopsHost)
+	if err != nil {
+		t.Fatalf("Error %s making new account", err)
+	}
+	queenClientInfo.Host = toznyCyclopsHost
+	identityServiceClient := New(queenClientInfo)
+
+	queenClientInfo.Host = e3dbAccountHost
+	accountToken := accountInfo.AccountServiceToken
+	queenAccountClient := accountClient.New(queenClientInfo)
+	registrationToken, err := test.CreateRegistrationToken(&queenAccountClient, accountToken)
+	if err != nil {
+		t.Fatalf("error %s creating account registration token using %+v %+v", err, queenAccountClient, registrationToken)
+	}
+	return identityServiceClient, registrationToken
+}
+
+func registerIdentity(t *testing.T, identityServiceClient E3dbIdentityClient, realmName string, registrationToken string) *RegisterIdentityResponse {
+	identityTag := uuid.New().String()
+	identityName := "Freud" + identityTag
+	identityEmail := "freud+" + identityTag + "@example.com"
+	identityFirstName := "Sigmund"
+	identityLastName := "Freud"
+	signingKeys, err := e3dbClients.GenerateSigningKeys()
+	if err != nil {
+		t.Fatalf("error %s generating identity signing keys", err)
+	}
+	encryptionKeyPair, err := e3dbClients.GenerateKeyPair()
+	if err != nil {
+		t.Fatalf("error %s generating encryption keys", err)
+	}
+	registerParams := RegisterIdentityRequest{
+		RealmRegistrationToken: registrationToken,
+		RealmName:              realmName,
+		Identity: Identity{
+			Name:        identityName,
+			Email:       identityEmail,
+			PublicKeys:  map[string]string{e3dbClients.DefaultEncryptionKeyType: encryptionKeyPair.Public.Material},
+			SigningKeys: map[string]string{signingKeys.Public.Type: signingKeys.Public.Material},
+			FirstName:   identityFirstName,
+			LastName:    identityLastName,
+		},
+	}
+	anonConfig := e3dbClients.ClientConfig{
+		Host: e3dbIdentityHost,
+	}
+	anonClient := New(anonConfig)
+	identity, err := anonClient.RegisterIdentity(testContext, registerParams)
+	if err != nil {
+		t.Fatalf("error %s registering identity using %+v %+v", err, anonClient, registerParams)
+	}
+	return identity
+}
+
+func uniqueString(prefix string) string {
+	rand.Seed(time.Now().UnixNano())
+
+	return fmt.Sprintf("%s%d", prefix, rand.Int())
+}
+
+func createRealmWithParams(t *testing.T, identityServiceClient E3dbIdentityClient, params CreateRealmRequest) *Realm {
+	realm, err := identityServiceClient.CreateRealm(testContext, params)
+	if err == nil {
+		return realm
+	}
+
+	t.Log("=================================")
+	t.Log("FAILED to create realm, RE-TRY...")
+	t.Log("=================================")
+
+	realm, err = identityServiceClient.CreateRealm(testContext, params)
```

Comment:
consider using Await or AwaitWithInterval methods here, example of this in action 
https://github.com/tozny/e3db-clients-go/blob/master/accountClient/accountClientV2_test.go#L181